basic/http/collect-user.js

```diff
@@ -25,13 +28,49 @@ class User {
 };
 
 const user = new User();
+
 router.get( '/users', async ctx => {
-  ctx.body = {data:user.data.map((e)=>{
+  ctx.body = {
+    data:user.data.map((e)=>{
       return {id:e.id,name:e.name}; 
     })
   };
+});
 
+router.post('/users',async ctx => {
+  
+  user.data.push(ctx.request.body);
+  
+  ctx.body = {
+    data:user.data.map((e)=>{
+      return {id:e.id,name:e.name}; 
+    })
+  };
 });
+
+router.patch('/users/:id',async ctx => {  
+
+  const updateUser = user.data;
+  for (const obj of updateUser) {
+    if (obj.updateUser.id === ctx.params.id) {
+      obj.name = ctx.request.body;
+
+    break;
+  }
+  const updated = user.data;
+  ctx.body = {
+    data:updated.map((e)=>{
+      return {id:e.id,name:e.name}; 
+    })
+  }
```

Comment:
We should only return a single object not an array. That object should be updated one